lib/big_query/base/timestamp.ex

```diff
@@ -41,17 +41,20 @@ defmodule BigQuery.Base.Timestamp do
     Regex.replace(~r/[ \n\r\t]+/, str, " ")
   end
 
-  def group({data, meta}, intv, group_by) do
+  def group({data, meta}, intv) do
     data = intv.rollup.range(intv.from, intv.to, false)
-           |> Enum.map(&({&1, find_times(&1, data, group_by)}))
+           |> insert_counts(data)
     {data, meta}
   end
 
-  defp find_times(time, data, "" <> group_by), do: find_times(time, data, String.to_atom(group_by))
-  defp find_times(time, data, group_by) do
-    time_unix = Timex.to_unix(time)
-    Enum.filter(data, fn(%{time: t}) -> Timex.to_unix(t) == time_unix end)
-    |> Enum.map(fn(d) -> {Map.get(d, group_by), d.count} end)
-    |> Map.new()
+  defp insert_counts([t1 | times], [%{time: t2, counts: counts} | rest] = all) do
+    if Timex.to_unix(t1) == Timex.to_unix(t2) do
```

Comment:
if the BQ sql is grouping by time, how would you have multiple rows with the same time?

Reply:
The first arg "times" is actually the "range" (complete list of timestamps) generated by `interval.rollup.range()`.

BigQuery may or may not return a row for every time in this range.  If an interval had 0 downloads in it, it wouldn't be in the resultset.  So this function makes sure every time is represented in the returned list.  Which is why if the range-time doesn't match the BQ-result-time ... I set an empty hash for that row and keep iterating.  (This all assumes BQ results come back ordered by timestamp ... which they do).

Reply:
makes sense, thanks!